src/bitmap/sorted_u16_vec.rs

```diff
@@ -0,0 +1,508 @@
+use crate::bitmap::bitmap_8k::{bit, key, Bitmap8K, BITMAP_LENGTH};
+use crate::bitmap::store::Store;
+use crate::bitmap::store::Store::Bitmap;
+use std::cmp::Ordering::*;
+use std::ops::{BitAnd, BitAndAssign, BitOr, BitXor, BitXorAssign, RangeInclusive, Sub, SubAssign};
+
+#[derive(Clone, Eq, PartialEq)]
+pub struct SortedU16Vec {
+    vec: Vec<u16>,
+}
+
+impl SortedU16Vec {
+    pub fn new() -> SortedU16Vec {
+        SortedU16Vec { vec: vec![] }
+    }
+
+    pub fn from_vec(vec: Vec<u16>) -> SortedU16Vec {
+        SortedU16Vec { vec }
+    }
```

Comment:
```suggestion
    pub fn from_sorted_vec(vec: Vec<u16>) -> SortedU16Vec {
        SortedU16Vec { vec }
    }
```

Reply:
I am not sure but don't we want this method to be `unsafe` as it can break the results, and we want the user of this method to be sure that the vector is valid?

Reply:
I like it and I love it.

Reply:
@Kerollmops The invariants of SortedU16Vec are not only sorted, but also unique. I'd welcome a suggestion for a better struct name.

Reply:
We could maybe use something like `U16VecSet`? and the method to construct it from a `Vec<u16>` can be `from_vec_unchecked`, this method **must** have indications on how to use it correctly (sorted and dedup).

I remember [an old crate I created with the same requirements](https://docs.rs/sdset/latest/sdset/), I called the type `SdSet`, for sorted and deduplicated.

Reply:
c694f54 and c130a9b

Note that the only usages of these unchecked creation are in SortedU16Vec::to_bitmap_store and Bitmap8k::to_array_store

Deserialization now uses the "try" versions, which will return an error in the case of invalid data. Huzzah!